rasa/core/test.py

```diff
@@ -600,18 +614,48 @@ def _get_e2e_entity_evaluation_result(
     return None
 
 
+def _get_predicted_action_name(
+    predicted_action: rasa.core.actions.action.Action,
+    partial_tracker: DialogueStateTracker,
+    expected_action_name: Text,
+) -> Text:
+    """Get the name of predicted action.
+
+    If the action is instance of `ActionRetrieveResponse`, we need to return full
+    action name with its retrieval intent (e.g. utter_faq/is-this-legit).
+    The only case when we should not do it is when an expected action given in
+    a test story is a retrieval action but it's not specified in the test story.
+    To illustrate this, we're basically avoiding this unnecessary mismatch:
+    utter_faq (expected) != utter_faq/is-this-legit (predicted).
+    In this case or if the action isn't instance of `ActionRetrieveResponse`,
+    the function returns only the action name (e.g. utter_faq).
```

Comment:
How are we computing metrics for these results? Do we still count "utter_faq/wrong-response" as correct regarding the intent classification then? Do we have separate metrics regarding the response selection? ... Doesn't really look like it or maybe I'm looking at the wrong modules.

Reply:
This code concerns only `run_core_test` part. Response selector is evaluated in the `run_nlu_test` and doesn't depend on this code. 

However, now that you pointed this out, I looked at the response selector evaluation and I think there are some problems... I'll look into it today.